### PR TITLE
[Yang] Fix Yang model of BGP Allowed Prefix

### PR DESCRIPTION
**Pull request by lizhijianrd**
#### Why I did it

Bugfix for Yang model of BGP Allowed Prefix.
1. Support optional `NEIGHBOR_TYPE` in key.
2. Support optional `le` and `ge` in prefixes_v4/prefixes_v6 list (e.g., `10.20.30.0/24 le 30`).

##### Work item tracking
- Microsoft ADO **(number only)**: 30001113

#### How I did it

Updated `sonic-bgp-allowed-prefix.yang`.
1. Define optional value `NEIGHBOR_TYPE` in key.
2. Define type `bgp-allowed-ipv4-prefix` and `bgp-allowed-ipv6-prefix` to support the optional suffix in prefixes_v4/prefixes_v6 list.

#### How to verify it

Verified by UT:

```
collected 870 items

tests/test_sonic_yang_models.py ..                                                        [  0%]
tests/yang_model_pytests/test_crm.py .................................................... [  6%]
......................................................................................... [ 16%]
......................................................................................... [ 26%]
......................................................................................... [ 36%]
......................................................................................... [ 47%]
......................................................................................... [ 57%]
.......................                                                                   [ 60%]
tests/yang_model_pytests/test_dash_crm.py ............................................... [ 65%]
......................................................................................... [ 75%]
......................................................................................... [ 85%]
......................................................................................... [ 96%]
........................                                                                  [ 98%]
tests/yang_model_pytests/test_smart_switch.py .........                                   [ 99%]
tests/yang_model_tests/test_yang_model.py .                                               [100%]

===================================== 870 passed in 33.19s ======================================
```

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: src/sonic-yang-models/yang-models/sonic-bgp-allowed-prefix.yang [hunk 1]
```diff
@@ -29,8 +29,125 @@ module sonic-bgp-allowed-prefix {
...
+
+         The string format is an inet:ipv6-prefix (defined in RFC 6991) value
+         appended with an optional suffix.
+         
+         The optional suffix consists of a string 'le' or 'ge' followed by a
+         number which is less than or equal to 128.";
+    }
+
     container sonic-bgp-allowed-prefix {
         container BGP_ALLOWED_PREFIXES {
```
Comment (wen587) [inline src/sonic-yang-models/yang-models/sonic-bgp-allowed-prefix.yang:76]: Please check sample YANG udpate PR here: https://github.com/sonic-net/sonic-buildimage/pull/19358. 
Need testcase to verify the changes.

Comment (lizhijianrd) [reply src/sonic-yang-models/yang-models/sonic-bgp-allowed-prefix.yang:76]: Hi @wen587, I've updated unittest. Can you please help review?

**Review by wen587 — APPROVED** [review]
lgttm

**Review by StormLiangMS — APPROVED** [review]
LGTM

**Comment by lizhijianrd** [comment]
Hi @bingwang-ms, can you please help backport 202405? Thanks.

**Pull request merged**